odl/solvers/smooth/gradient.py

```diff
@@ -8,9 +8,8 @@
 
 """Gradient-based optimization schemes."""
 
-# Imports for common Python 2/3 codebase
 from __future__ import print_function, division, absolute_import
-
+from builtins import range
```

Comment:
Have we settled this discussion before somewhere? I admit that these imports are not necessary, they just make the code behave the same in Python 2 and 3. 

Certainly `range` and `zip` with generator vs. list behavior makes virtually no difference.

`object` is necessary when subclassing and implementing custom iterators. This can be relevant, so I'm inclined to keeping that import.

For `int` there is potential for screw-up when doing stuff like `np.ones(3, dtype=int)` in Python 2, so these should be imported locally only when large integers are expected (e.g. `size`).

`str` may be a matter of taste, but I guess in order to keep printed stuff without `u` or `b` it shouldn't be there either 🤔 

Anything else I forgot?

Reply:
Well this thinking of us enforcing that stuff should work the same is (imo) somewhat outside of the ODL idea of "being easy to leave ODL", i.e. if ODL objects are "contaminated" by using slightly different ints etc then interoparability will suffer.

My suggestion is that we only do these things when it is rather critical, and not just because we want to potentiall fix some very special case.

Reply:
Ok, but I'll keep using new int for size-like variables where overflow of 32 bit integers is not so unlikely.